### PR TITLE
IC-1992: Allow users to download list of structured interventions

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

Adds a section to Find where users can download an .xlsx or .pdf version of a list of Structured Interventions.

This PR configures `git-lfs` to allow GitHub to more efficiently host our pdf files. It will require developers to install the package to be able to commit locally, as it runs as a pre-commit hook. See commit message in #694aa72 for more information.
 
## What is the intent behind these changes?

To give PPs a place to view all structured interventions that are currently unsupported by the service.

## Screenshot

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +77,15 @@
     "delay": 2500,
     "ext": "js,json,html,njk"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
```
Comment (lawrence-forooghian) [inline package.json:82]: In #532 we upgraded to Husky 7, which changes the way hooks are defined. They're now defined as scripts like `pre-commit` in the repo's `.husky` directory. So I reckon that you'll need to take these out of here and put them in` .husky` (perhaps you can move the contents of the `hooks` directory into a directory called e.g. `.husky/git-lfs`, too?)

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 2]
```diff
@@ -59,7 +60,11 @@ export default class ProbationPractitionerReferralsController {
     const { accessToken } = res.locals.user.token
 
     const existingDraftReferrals = await this.interventionsService.getDraftReferralsForUserToken(accessToken)
-    const presenter = new FindStartPresenter(existingDraftReferrals)
+
+    const downloadFileInfo = await FileUtils.fileInformation(
+      'assets/downloads/Structured interventions list v1_3_3.xlsx'
+    )
+    const presenter = new FindStartPresenter(existingDraftReferrals, downloadFileInfo)
```
Comment (lawrence-forooghian) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:67]: It feels a bit off for me to have a controller passing a formatted string to a presenter. Perhaps it would be more appropriate for the controller to be responsible for fetching the file's size as a `number` of bytes, and then the presenter would do the formatting.

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 3]
```diff
@@ -59,7 +60,11 @@ export default class ProbationPractitionerReferralsController {
     const { accessToken } = res.locals.user.token
 
     const existingDraftReferrals = await this.interventionsService.getDraftReferralsForUserToken(accessToken)
-    const presenter = new FindStartPresenter(existingDraftReferrals)
+
+    const downloadFileInfo = await FileUtils.fileInformation(
```
Comment (lawrence-forooghian) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:64]: Can we find a way to avoid the duplication of this filename between here and `FindStartPresenter`?

File: server/utils/fileUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import FileUtils from './fileUtils'
+
+describe(FileUtils, () => {
+  describe('.fileInformation', () => {
+    it('returns the file type and size in KB, rounded to one decimal place', async () => {
+      expect(await FileUtils.fileInformation('testutils/test_logo.png')).toEqual('PNG, 10.9KB')
```
Comment (lawrence-forooghian) [inline server/utils/fileUtils.test.ts:6]: Perhaps this could go in a separate commit?

File: .gitattributes [hunk 5]
```diff
@@ -0,0 +1 @@
+*.pdf filter=lfs diff=lfs merge=lfs -text
```
Comment (sldblog) [inline .gitattributes:1]: 😍

File: server/utils/fileUtils.test.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import FileUtils from './fileUtils'
+
+describe(FileUtils, () => {
+  describe('.fileInformation', () => {
+    it('returns the file type and size in KB, rounded to one decimal place', async () => {
```
Comment (lawrence-forooghian) [inline server/utils/fileUtils.test.ts:5]: I reckon we can return this in bytes; converting to KB and rounding can probs be considered a formatting concern.

File: assets/sass/components/_structured-interventions-downloads.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
+.download-section {
```
Comment (lawrence-forooghian) [inline assets/sass/components/_structured-interventions-downloads.scss:1]: We probably need some namespacing here.

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 8]
```diff
@@ -59,7 +60,19 @@ export default class ProbationPractitionerReferralsController {
...
 
     const existingDraftReferrals = await this.interventionsService.getDraftReferralsForUserToken(accessToken)
-    const presenter = new FindStartPresenter(existingDraftReferrals)
+
+    const structuredInterventionsFileDownloadPaths = {
+      xlsx: 'assets/downloads/Structured interventions list v1_3_3.xlsx',
+      pdf: 'assets/downloads/Structured interventions list v1_3_3.pdf',
+    }
+
+    const downloadFileSize = await FileUtils.fileSizeInKB(structuredInterventionsFileDownloadPaths.xlsx)
```
Comment (lawrence-forooghian) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:69]: I wonder what these relative paths will be resolved against here. Probably the process's current working directory. I think it would be good to be explicit here, I'll take a look at how these paths get configured for the assets.

Comment (lawrence-forooghian) [reply server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:69]: Oh, looks like they resolve relative to `process.cwd()` there, too. OK, will leave it.
